crates/sui-json-rpc-api/src/rpc_types.rs

```diff
@@ -1292,3 +1292,24 @@ impl From<TypeTag> for SuiTypeTag {
         Self(format!("{}", tag))
     }
 }
+
+#[derive(Serialize, Deserialize, JsonSchema)]
+pub enum RPCTransactionRequestParams {
+    TransferCoinRequestParams(RPCTransferCoinRequestParams),
+    MoveCallRequestParams(RPCMoveCallRequestParams),
+}
+
+#[derive(Serialize, Deserialize, JsonSchema)]
+pub struct RPCTransferCoinRequestParams {
+    pub recipient: SuiAddress,
+    pub object_id: ObjectID,
+}
+
+#[derive(Serialize, Deserialize, JsonSchema)]
+pub struct RPCMoveCallRequestParams {
+    pub package_object_id: ObjectID,
+    pub module: String,
+    pub function: String,
+    pub type_arguments: Vec<SuiTypeTag>,
+    pub arguments: Vec<SuiJsonValue>,
+}
```

Comment:
lets add `#[serde(rename_all = "camelCase")]` to both struct to make the parameter more json friendly

Reply:
also we can add `#[serde(default)]` to type_arg and args to allow omitting the input if they are empty